### PR TITLE
Add ArrayDup from raw pointer

### PR DESCRIPTION
**Pull request by liu15**

### CODE REVIEW

**Review by adayton1 — APPROVED** [review]
LGTM

File: src/care/algorithm_impl.h [hunk 1]
```diff
@@ -1515,21 +1515,79 @@ CARE_INLINE void ArrayCopy(RAJA::seq_exec,
    } CARE_SEQUENTIAL_LOOP_END
 }
 
+/************************************************************************
+ * Function  : ArrayCopy
```
Comment (dtaller) [inline src/care/algorithm_impl.h:1519]: Documentation: change "ArrayCopy" to "ArrayDup" to match the function name.

File: src/care/algorithm_impl.h [hunk 2]
```diff
@@ -1515,21 +1515,79 @@ CARE_INLINE void ArrayCopy(RAJA::seq_exec,
...
+ * Purpose   : Duplicates a ManagedArray.
+ * ************************************************************************/
+template <typename T>
+CARE_INLINE care::host_device_ptr<T> ArrayDup(care::host_device_ptr<const T> from, int len)
+{
+   return ArrayDup<T>(RAJAExec {}, from, len);
+}
+
+/************************************************************************
+ * Function  : ArrayCopy
```
Comment (dtaller) [inline src/care/algorithm_impl.h:1530]: Documentation: change "ArrayCopy" to "ArrayDup" to match the function name.

File: src/care/algorithm_impl.h (hunk 1 above)
Comment (adayton1) [reply src/care/algorithm_impl.h]: Fixed.

File: src/care/algorithm_impl.h (hunk 2 above)
Comment (adayton1) [reply src/care/algorithm_impl.h]: Fixed.

**Review by dtaller — APPROVED** [review]
Thanks for the fixes, looks good!

**Pull request merged**